shared/settings/chat/index.tsx

```diff
@@ -70,9 +93,102 @@ class Chat extends React.Component<Props, State> {
+          <Kb.Text type="Header">Contact settings</Kb.Text>
+        </Kb.Box2>
+        <Kb.Box2 direction="vertical" fullWidth={true}>
+          <Kb.Checkbox
+            label="Only let someone message me or add me to a team if:"
+            onCheck={() =>
+              this.setState(prevState => ({contactSettingsEnabled: !prevState.contactSettingsEnabled}))
+            }
+            checked={!!this.state.contactSettingsEnabled}
+            disabled={this.props.contactSettingsEnabled === undefined}
+          />
+          {!!this.state.contactSettingsEnabled && (
+            <>
+              <Kb.Checkbox label="I follow them, or..." checked={true} onCheck={null} />
+              <Kb.Checkbox
+                label="I follow someone who follows them, or..."
+                onCheck={checked =>
+                  this.setState({
+                    contactSettingsIndirectFollowees: checked,
+                  })
+                }
+                checked={!!this.state.contactSettingsIndirectFollowees}
+              />
+              <Kb.Checkbox
+                label="They're in one of these teams with me:"
+                onCheck={checked => this.setState({contactSettingsTeamsEnabled: checked})}
+                checked={!!this.state.contactSettingsTeamsEnabled}
+                disabled={false}
+              />
+              {this.state.contactSettingsTeamsEnabled && (
+                <Kb.ScrollView style={styles.teamList}>
+                  {this.props.teamDetails.map(teamDetails => (
+                    <TeamRow
+                      checked={this.state.contactSettingsSelectedTeams[teamDetails.id]}
+                      key={teamDetails.id}
+                      isOpen={teamDetails.isOpen}
+                      membercount={teamDetails.memberCount}
+                      name={teamDetails.teamname}
+                      onCheck={(checked: boolean) =>
+                        this.setState(prevState => ({
+                          contactSettingsSelectedTeams: {
+                            ...prevState.contactSettingsSelectedTeams,
+                            [teamDetails.id]: checked,
+                          },
+                        }))
+                      }
+                    />
+                  ))}
+                </Kb.ScrollView>
+              )}
+            </>
+          )}
+          <Kb.Box2 direction="vertical" gap="tiny">
+            <Kb.WaitingButton
+              onClick={() =>
+                this.props.onContactSettingsSave(
+                  !!this.state.contactSettingsEnabled,
+                  !!this.state.contactSettingsIndirectFollowees,
+                  !!this.state.contactSettingsTeamsEnabled,
+                  this.state.contactSettingsSelectedTeams
+                )
+              }
+              label="Save"
+              style={styles.save}
+              waitingKey={Constants.contactSettingsWaitingKey}
+            />
```

Comment:
I think this save button is a bit smushed between the post link previews and the teams list above. 
Would it be possible to add some more space above/below this button?